### ISSUE TITLE
govc: Attach Disk to SCSI Controller

### ISSUE THREAD

**Issue by gothicx** [vmware/govmomi#2117]
Hi,

It would be useful to have a device.scsi.attach to attach a disk to a scsi controller.

govc device.scsi.attach -vm $vm pvscsi-1001

Thank you.

**Comment by dougm** [comment]
I think we could add a `-controller` flag to `vm.disk.change` to support this. What do you think?

**Comment by gothicx** [comment]
That's ok for me. Thank you !

**Comment by jbertozzi** [comment]
Hello,

It seems the code has been modifed to add this functionnality.

However, it seems not to be working (using a VSAN datastore):

`govc vm.disk.create -controller pvscsi-1002 -vm=myvm01 -name myvm01_2.vmdk -size 2G -ds dsvsan -mode=independent_persistent -eager -thick -dump
&types.LocalizedMethodFault{
    Fault: &types.SystemError{
        RuntimeFault: types.RuntimeFault{
            MethodFault: types.MethodFault{
                FaultCause:   (*types.LocalizedMethodFault)(nil),
                FaultMessage: []types.LocalizableMessage{
                    {
                        Key:     "vob.fssvec.CreateFile.file.failed",
                        Arg:     nil,
                        Message: "File system specific implementation of CreateFile[file] failed",
                    },
                    {
                        Key:     "vob.fssvec.CreateFile.file.failed",
                        Arg:     nil,
                        Message: "File system specific implementation of CreateFile[file] failed",
                    },
                    {
                        Key:     "vob.fssvec.CreateFile.file.failed",
                        Arg:     nil,
                        Message: "File system specific implementation of CreateFile[file] failed",
                    },
                },
            },
        },
        Reason: "Error creating disk Function not implemented",
    },
    LocalizedMessage: "A general system error occurred: Error creating disk Function not implemented",
}
`
Should I open a new issue for this?

**Comment by embano1** [comment]
Yes, if this issue is not correctly reflecting the bug you're seeing, please open a new one.

**Comment by gothicx** [comment]
@jbertozzi This bug was reported to add a controller to an existing disk and not while creating a new VM, so you probably want to report a new bug.

**Comment by jbertozzi** [comment]
New issue opened:

https://github.com/vmware/govmomi/issues/2568
